ssz/sedes/list.py

```diff
@@ -0,0 +1,96 @@
+from collections.abc import (
+    Iterable,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes import (
+    address,
+    boolean,
+    hash32,
+    uint32,
+)
+
+
+class List:
+    """
+    A sedes for lists.
+
+    WARNING: Avoid sets if possible, may not always lead to expected results
+    (This is because iteration in sets doesn't always happen in the same order)
+    """
+    LENGTH_BYTES = 4
+
+    def __init__(self, element_sedes):
+        # This sedes object corresponds to each element of the iterable
+        self.element_sedes = element_sedes
+
+    def serialize(self, val):
+        if not isinstance(val, Iterable):
```

Comment:
If you change this to use `collections.abc.Sequence` then it will disallow non-ordered iterable types like `set` or `dict`.  In addition, you should potentially disallow all of `bytes,bytearray,str` since they will be *interpreted* as an iterable of their individual characters which is almost definitely not the desired/expected behavior.

Reply:
Agree with the string like checks. But I'd rather keep the `Iterable` as `Sequence` would exclude generators which might be nice to have in some cases.